ranger/config.go

```diff
@@ -0,0 +1,50 @@
+package ranger
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/xy-planning-network/trails"
+	"github.com/xy-planning-network/trails/http/middleware"
+	"github.com/xy-planning-network/trails/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type Config[User RangerUser] struct {
+	// NOTE(dlk): Ranger can accept a type parameter also.
+	// Config was chosen to minimize proliferating generic type parameters
+	// in all Ranger methods or references to Ranger.
+	// Config ought to be restricted to New.
+}
+
+// defaultUserStore constructs a function matching the signature of middleware.UserStorer.
+// This function pulls the User from the db by ID,
+// preloading all top-level associations.
+func (Config[User]) defaultUserStore(db postgres.DatabaseService) middleware.UserStorer {
```

Comment:
This is a brilliant + simple way to get the fetching of the user into trails. Cool application of generics here 🥳